client/src/AppBundle/Controller/Admin/Client/ReportController.php

```diff
@@ -436,7 +441,11 @@ public function manageConfirmAction(Request $request, $id)
             $this->restClient->put('report/' . $report->getId(), $report, ['report_type', 'report_due_date']);
 
             if ($form->has('confirm') && $form['confirm']->getData() === 'yes' && $report->isSubmitted()) {
-                $this->unsubmitReport($report);
+                $this->reportApi->unsubmit(
+                    $report,
+                    $this->get('security.token_storage')->getToken()->getUser(),
```

Comment:
All controllers have a built in `getUser()` function which saves us having to dip in and grab a service from the container:

```suggestion
                    $this->getUser(),
```

Reply:
I thought this but php storm was saying nay (expecting entity rather than interface error) but it seems to work anyway. Cheers.

Reply:
Yeah I get that too - its one of those times where the IDE isn't quite smart enough to know we're going to get a proper entity user object back.